src/hotspot/share/cds/archiveHeapWriter.cpp

```diff
@@ -79,22 +79,20 @@ void ArchiveHeapWriter::init() {
   if (HeapShared::can_write()) {
     Universe::heap()->collect(GCCause::_java_lang_system_gc);
 
-    _buffer_offset_to_source_obj_table = new BufferOffsetToSourceObjectTable();
+    _buffer_offset_to_source_obj_table = new BufferOffsetToSourceObjectTable(36137, 1 * M);
```

Comment:
Is there a variable we can use instead of the hard coded value 36137? It isn't immediately clear why this value is used and I think a named variable would clear that up.

Reply:
I changed to something like this:

`new BufferOffsetToSourceObjectTable(/*size*/36137, /*max size*/1 * M)`

Reply:
My main confusion is why 36137 was chosen specifically. I figure it's a prime number to set as the initial size of the table?

Reply:
Yes, it was chosen because it's a prime number. I added comments about that to the code.